### PR TITLE
Add core concepts - workflow types

### PR DESCRIPTION
**Pull request by anludke**
This PR introduces a new 'Core Concepts' section to the documentation, which begins with a comprehensive explanation and examples of workflow types in the Orchestrator. This addition addresses a previously missing piece of documentation, aiming to provide essential guidance for developers creating assessment workflows from the Basic workflow bootstrap project. 

The output of this PR should look like:

Screenshot 2024-05-07 at 2 33 43 PM

Screenshot 2024-05-07 at 2 34 00 PM

Screenshot 2024-05-07 at 2 34 09 PM

### CODE REVIEW

**Comment by batzionb** [comment]
We currently have these almost completely empty pages
https://www.parodos.dev/docs/serverless-workflows/assessment/
and
https://www.parodos.dev/docs/serverless-workflows/infrastructure/
Maybe it would be better to put this content in these places? WDYT?

**Review by pkliczewski — COMMENTED** [review]
/lgtm

I agree with @batzionb it would be great to link this content there as well

**Comment by anludke** [comment]
@batzionb @pkliczewski I've explicitly positioned the content at the top in a new "core concepts" section to emphasize its importance, as this specific workflow type has not been covered elsewhere, despite the examples provided in subsequent sections.
Although the infrastructure and assessment sections may appear empty for now, they were intentionally created to separate workflows by type for clarity.
By prioritizing this content upfront, I aim to establish a solid foundation for understanding this workflow type and facilitate the development of workflows for the Orchestrator.

File: content/docs/core-concepts/workflow-types/_index.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+title: "Workflow Types"
+date: 2024-05-07
+---
+
+The *Orchestrator* features two primary workflow categories:
+- *Infrastructure workflows*: focus on automating infrastructure-related tasks
+- *Assessment workflows*: focus on evaluating and analyzing data to suggest suitable infrastructure workflow options for subsequent execution
+
+### Infrastructure workflow
+In Orchestration, an infrastructure refers to a workflow that executes a sequence of operations based on user input (optional) and generates output (optional) without requiring further action.
```
[the remark this answers is not in the document]
Comment (anludke) [reply content/docs/core-concepts/workflow-types/_index.md:11]: > is `Orchestration` ok? The capital letter makes me think it's a specific component o product

Good catch!

**Review by gciavarrini — APPROVED** [review]
lgtm

**Pull request merged**